### PR TITLE
external/esp_idf_port: Add retry times when esp32 wifi disconnected.

### PR DESCRIPTION
**Pull request by ChinaTizenRT**
In some situation, esp32 wifi will connect AP failed for the first time,
Giving chance to reconnect will improve this case.
@pilseob.kang
@sh.byeon

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [8286386b48a8ea7d960028fe27fb85f0149fdbdb] -

**Comment by seinfra** [comment]
Target : [8286386b48a8ea7d960028fe27fb85f0149fdbdb] -

File: external/esp_idf_port/esp32/esp32_wifi_utils.c [hunk 1]
```diff
@@ -245,13 +246,20 @@ static esp_err_t event_handler(void *ctx, system_event_t *event)
 		s_retry_num++;
 		int reason = event->event_info.disconnected.reason;
 		ESP_LOGI(TAG, "[%s] connect to the AP %s fail %d: %d; %d\n", __func__, event->event_info.disconnected.ssid, s_retry_num, event->event_info.disconnected.reason, isStaConnected);
-    
+
```
Comment (xiarihf) [inline external/esp_idf_port/esp32/esp32_wifi_utils.c]: Check coding rules.

Comment (SRCN-LiQin) [inline external/esp_idf_port/esp32/esp32_wifi_utils.c]: new space is added in this line

File: external/esp_idf_port/esp32/esp32_wifi_utils.c [hunk 2]
```diff
@@ -245,13 +246,20 @@ static esp_err_t event_handler(void *ctx, system_event_t *event)
...
 		ESP_LOGI(TAG, "[%s] connect to the AP %s fail %d: %d; %d\n", __func__, event->event_info.disconnected.ssid, s_retry_num, event->event_info.disconnected.reason, isStaConnected);
-    
+        
         /*give chance to other thread to continue*/
         sleep(1);
 		if (s_retry_num > 0) {
 			ESP_LOGI(TAG, "g_cbk.sta_connected fail\n");
-			g_cbk.sta_connected(WIFI_UTILS_FAIL, NULL);
-			s_retry_num = 0;
+            if (s_retry_num < WIFI_CONN_RETRY) {
```
Comment (SRCN-LiQin) [inline external/esp_idf_port/esp32/esp32_wifi_utils.c]: It's not align with code above

File: external/esp_idf_port/esp32/esp32_wifi_utils.c [hunk 3]
```diff
@@ -245,13 +246,20 @@ static esp_err_t event_handler(void *ctx, system_event_t *event)
...
+                esp_wifi_connect();
+            }
+
+            if (s_retry_num == WIFI_CONN_RETRY) {
```
Comment (SRCN-LiQin) [inline external/esp_idf_port/esp32/esp32_wifi_utils.c]: how about else if or else for this? avoid unnecessary condition check here

File: external/esp_idf_port/esp32/esp32_wifi_utils.c [hunk 4]
```diff
@@ -245,13 +246,20 @@ static esp_err_t event_handler(void *ctx, system_event_t *event)
...
+                g_cbk.sta_connected(WIFI_UTILS_FAIL, NULL);
+                s_retry_num = 0;
+            }
+
 		} else {
 			ESP_LOGI(TAG,"g_cbk.sta_disconnected\n");
```
Comment (zhouxinhe) [inline external/esp_idf_port/esp32/esp32_wifi_utils.c]: is this code reachable?

File: external/esp_idf_port/esp32/esp32_wifi_utils.c [hunk 5]
```diff
@@ -245,13 +246,20 @@ static esp_err_t event_handler(void *ctx, system_event_t *event)
 		s_retry_num++;
 		int reason = event->event_info.disconnected.reason;
 		ESP_LOGI(TAG, "[%s] connect to the AP %s fail %d: %d; %d\n", __func__, event->event_info.disconnected.ssid, s_retry_num, event->event_info.disconnected.reason, isStaConnected);
-    
+        
         /*give chance to other thread to continue*/
         sleep(1);
```
Comment (zhouxinhe) [inline external/esp_idf_port/esp32/esp32_wifi_utils.c]: Please also check alignment here.

File: external/esp_idf_port/esp32/esp32_wifi_utils.c (hunk 2 above)
Comment (xixidodo) [inline external/esp_idf_port/esp32/esp32_wifi_utils.c]: please check if below scenario is possible：
after 'wifi_utils_disconnect_ap' invoked,  'SYSTEM_EVENT_STA_DISCONNECTED' fired multiple times from driver due to somewhat reason.

 if so, then ap is retried wrongly.

File: external/esp_idf_port/esp32/esp32_wifi_utils.c [hunk 6]
```diff
@@ -245,13 +246,20 @@ static esp_err_t event_handler(void *ctx, system_event_t *event)
...
 		if (s_retry_num > 0) {
 			ESP_LOGI(TAG, "g_cbk.sta_connected fail\n");
```
Comment (xixidodo) [inline external/esp_idf_port/esp32/esp32_wifi_utils.c]: please check alignment

File: external/esp_idf_port/esp32/esp32_wifi_utils.c [hunk 7]
```diff
@@ -245,13 +246,20 @@ static esp_err_t event_handler(void *ctx, system_event_t *event)
...
+            if (s_retry_num < WIFI_CONN_RETRY) {
+                esp_wifi_connect();
+            }
+
+            if (s_retry_num == WIFI_CONN_RETRY) {
+                g_cbk.sta_connected(WIFI_UTILS_FAIL, NULL);
+                s_retry_num = 0;
+            }
+
 		} else {
```
Comment (xiarihf) [inline external/esp_idf_port/esp32/esp32_wifi_utils.c]: Usless code. s_retry_num could not be negative number, it wiill be 0~5.

File: external/esp_idf_port/esp32/esp32_wifi_utils.c [hunk 8]
```diff
@@ -56,6 +56,7 @@
 #include "nvs_flash.h"
 #include "esp_wifi_internal.h"
 
+#define WIFI_CONN_RETRY  5
```
Comment (drashti304) [inline external/esp_idf_port/esp32/esp32_wifi_utils.c]: please make it indented to below defined macros

Comment (ChinaTizenRT) [reply external/esp_idf_port/esp32/esp32_wifi_utils.c]: tks, fixed

File: external/esp_idf_port/esp32/esp32_wifi_utils.c (hunk 1 above)
Comment (ChinaTizenRT) [reply external/esp_idf_port/esp32/esp32_wifi_utils.c]: tks, fixed

Comment (ChinaTizenRT) [reply external/esp_idf_port/esp32/esp32_wifi_utils.c]: tks, fixed

File: external/esp_idf_port/esp32/esp32_wifi_utils.c (hunk 5 above)
Comment (ChinaTizenRT) [reply external/esp_idf_port/esp32/esp32_wifi_utils.c]: tks, fixed

File: external/esp_idf_port/esp32/esp32_wifi_utils.c (hunk 6 above)
Comment (ChinaTizenRT) [reply external/esp_idf_port/esp32/esp32_wifi_utils.c]: tks, fixed

File: external/esp_idf_port/esp32/esp32_wifi_utils.c (hunk 2 above)
Comment (ChinaTizenRT) [reply external/esp_idf_port/esp32/esp32_wifi_utils.c]: tks, fixed

File: external/esp_idf_port/esp32/esp32_wifi_utils.c (hunk 7 above)
Comment (ChinaTizenRT) [reply external/esp_idf_port/esp32/esp32_wifi_utils.c]: it is signed int, and initial as -1 is used for  to differ case "disconnect ap" from "connect ap fail".

File: external/esp_idf_port/esp32/esp32_wifi_utils.c (hunk 4 above)
Comment (ChinaTizenRT) [reply external/esp_idf_port/esp32/esp32_wifi_utils.c]: yes, station disconnect ap activly case.

File: external/esp_idf_port/esp32/esp32_wifi_utils.c (hunk 3 above)
Comment (ChinaTizenRT) [reply external/esp_idf_port/esp32/esp32_wifi_utils.c]: tks, using else looks logic clear.

File: external/esp_idf_port/esp32/esp32_wifi_utils.c (hunk 2 above)
Comment (ChinaTizenRT) [reply external/esp_idf_port/esp32/esp32_wifi_utils.c]: tks, reset s_retry_num to fix this case

**Comment by seinfra** [comment]
Target : [124c6cef5e3179a87e92c94ae8db4580ca31b5e4] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [124c6cef5e3179a87e92c94ae8db4580ca31b5e4] - Code Rule Check (C++) OK.

File: external/esp_idf_port/esp32/esp32_wifi_utils.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (ChinaTizenRT) [reply external/esp_idf_port/esp32/esp32_wifi_utils.c]: argee, this issue is already fixed, tks.

**Pull request merged**